File: app/src/main/java/org/pcsoft/app/jmc/ui/windows/MainWindow.java
package org.pcsoft.app.jmc.ui.windows;

import javafx.scene.Scene;
import javafx.stage.Stage;
import org.pcsoft.framework.jfex.mvvm.Fxml;

public class MainWindow extends Stage {

    private final MainWindowViewModel viewModel;
    private final MainWindowView viewController;

    public MainWindow() {
        setTitle("Java Media Center");

        final var tuple = Fxml.from(MainWindowView.class).load();
        setScene(new Scene(tuple.getView(), 1024, 800));

        viewModel = tuple.getViewModel();
        viewController = tuple.getViewController();
    }
}

File: app/src/main/kotlin/org/pcsoft/app/jmc/ui/windows/MainWindow.kt
package org.pcsoft.app.jmc.ui.windows

import javafx.scene.Scene
import javafx.stage.Stage
import org.pcsoft.framework.jfex.mvvm.Fxml

class MainWindow : Stage() {

    private val controller: MainWindowView
    private val model: MainWindowViewModel

    init {
        title = "Java Media Center"

        val tuple = Fxml.from(MainWindowView::class.java).load()
        scene = Scene(tuple.view, 1024.0, 800.0)

        model = tuple.viewModel
        controller = tuple.viewController
    }

}